### PR TITLE
Add a Docker image with an Ubuntu 14.04 ARM rootfs on an Ubuntu 16.04 base image

### PR DESCRIPTION
**Pull request by jkoritzinsky**
I was talking with @janvorli about dotnet/coreclr#23999 and he requested that we keep building CoreCLR against the version of libc in Ubuntu 14.04.

This PR adds a Docker image that generates a 14.04 rootfs to satisfy that request, while running on Ubuntu 16.04 to enable us to move off of the EOL'd Ubuntu 14.04 version.

### CODE REVIEW

File: src/ubuntu/16.04/cross/14.04/hooks/pre-build [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+$(dirname ${BASH_SOURCE[0]})/../../../build-scripts/build-rootfs.sh ubuntu-14.04 trusty
```
Comment (janvorli) [inline src/ubuntu/16.04/cross/14.04/hooks/pre-build]: I've just created a PR for updating this script in the original 14.04 form (https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/145). Can you please copy the change here?

File: src/ubuntu/16.04/cross/14.04/hooks/post-build [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../build-scripts/build-rootfs-cleanup.sh
```
Comment (MichaelSimons) [inline src/ubuntu/16.04/cross/14.04/hooks/post-build]: Nit - missing line ending

**Comment by jkoritzinsky** [comment]
@MichaelSimons I'm not sure how to fix the current failure. Any suggestions?

**Comment by MichaelSimons** [comment]
@jkoritzinsky - The problem is the logic to build the rootfs still has a [dependency on 14.04 images](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/master/src/ubuntu/build-scripts/build-rootfs.sh#L13) given the [way it is invoked](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/146/files#diff-684ab3e1fe78534a4dbcde5d76b7c215R3).  Because this dependency is not expressed within the manifest/Dockerfile FROM instructions, the `dotnet-buildtools/prereqs:ubuntu-14.04-crossdeps` does not exist when building the `ubuntu-16.04-cross-14.04` image thus breaking the build.

I'm not fully understanding the intent of this change.  We want to remove the dependency on 14.04 but still want to build targeting 14.04 ARM?

cc @mthalman

**Comment by jkoritzinsky** [comment]
Yes. Although we are not explicitly supporting Ubuntu 14.04 on ARM, @janvorli [expressed intent](https://github.com/dotnet/coreclr/pull/23999#issuecomment-483390604) to not explicitly break support for running .NET Core 3.0 on ARM distros with a libc version before 2.23.

My goal was two-pronged: move off unsupported Ubuntu 14.04 for our build images so that we aren't building on unsupported operating systems, and move to Ubuntu 16.04 to allow CoreCLR to use versions of CMake that are all within the same major version number.

We agreed that building an image that would run on Ubuntu 16.04 but allow us to target 14.04 ARM (with its libc version) would enable us to support both of our goals.

**Comment by MichaelSimons** [comment]
I think more changes are going to be necessary to achieve the goal of moving off of the unsupported Ubuntu 14.04 because with these changes, we are still relying on the use of 14.04 to generate the rootfs.  Can we run `build-rootfs.sh` on 16.04 while specifying the `trusty` toolset?

**Comment by janvorli** [comment]
> Can we run `build-rootfs.sh` on 16.04 while specifying the `trusty` toolset?

Yes, there is no problem in that. The host system and the system in the rootfs versions are independent

**Comment by jkoritzinsky** [comment]
I didn't realize that those two parameters were independent. I'll try that out.

**Comment by jkoritzinsky** [comment]
That got a lot further, but it looks like it's getting an `Exec format error` on cleanup so something about the toolchain isn't getting set up right.

**Comment by MichaelSimons** [comment]
I'm going to re-queue because that may be a Windows Docker issue.

File: src/ubuntu/16.04/cross/14.04/hooks/post-build [hunk 3]
```diff
@@ -0,0 +1 @@
+../../../../build-scripts/build-rootfs-cleanup.sh
```
Comment (janvorli) [inline src/ubuntu/16.04/cross/14.04/hooks/post-build]: It seems this should also start with `$(dirname ${BASH_SOURCE[0]})/`

**Comment by jkoritzinsky** [comment]
Looks like the build passed!

**Comment by jkoritzinsky** [comment]
I don't have permission to merge in this repo so can someone merge?

**Comment by MichaelSimons** [comment]
The following images were built as a result of merging this PR:

mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-coredeps-3e800f1-20190528233931
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-crossdeps-cfdd435-20190528233931
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-cross-arm64-cfdd435-20190528233931
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-cross-arm64-alpine-406629a-20190528233931
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-cross-23cacb0-20190528233931
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-cross-14.04-23cacb0-20190528233931
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-23cacb0-20190528233934
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-debpkg-cfdd435-20190528233934
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-mlnet-207e097-20190528233934
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-arm64v8-3e800f1-20190528233801
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-helix-arm64v8-09ca40b-20190528233801
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-arm32v7-0dbbad8-20190528233802
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-arm32v7-0dbbad8-20190528233802
mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-16.04-helix-arm32v7-09ca40b-20190528233803

**Pull request merged**